android/modules/ui/src/js/listview.js

```diff
@@ -80,13 +80,29 @@ exports.bootstrap = function(Titanium) {
 	
 	//convert name of UI elements into a constructor function.
 	//I.e: lookup("Titanium.UI.Label") returns Titanium.UI.createLabel function
-	function lookup(name) {
-		var lastDotIndex = name.lastIndexOf('.');
-		var proxy = eval(name.substring(0, lastDotIndex));
-		if (typeof(proxy) == undefined) return;
+	function lookup(namespace) {
 
-		var proxyName = name.slice(lastDotIndex + 1);
-		return proxy['create' + proxyName];
+		// handle Titanium widgets
+		if (/^(Ti|Titanium)/.test(namespace)) {
+			const namespaceIndex = namespace.lastIndexOf('.'),
+				  proxyName = namespace.slice(namespaceIndex + 1),
+				  parentNamespace = namespace.substring(0, namespaceIndex),
+				  parentProxy = eval(parentNamespace);
+	
+			if (parentProxy) {
+				return parentProxy['create' + proxyName];
+			}
+	
+		// handle Alloy widgets
+		} else {
+			const widget = Module.main.require('/alloy/widgets/' + namespace + '/controllers/widget');
+			if (widget) {
```

Comment:
We should handle plain CommonJS modules here as well? For iOS, we use a [fallback case](https://github.com/appcelerator/titanium_mobile/pull/10131/files#diff-e577c4c73d4af3fdf412c02fff71ed4fR2955) to do this. We require common-js modules to export a view via `getView` as well, which is a suitable requirement to have a formal way of knowing where the main view was exported.

Reply:
@garymathews Ping! The iOS PR was just merged.